eng/tox/create_package_and_install.py

```diff
@@ -188,16 +202,39 @@ def build_and_discover_package(setuppy_path, dist_dir, target_setup, package_typ
                     logging.info("Installing {w} from directory".format(w=built_package))
                 # it does't exist, so we need to error out
                 else:
-                    logging.error(
-                        "{w} not present in the prebuilt package directory. Exiting.".format(
-                            w=built_package
-                        )
-                    )
+                    logging.error("{w} not present in the prebuilt package directory. Exiting.".format(w=built_package))
                     exit(1)
             else:
                 built_pkg_path = os.path.abspath(os.path.join(args.distribution_directory, built_package))
                 logging.info("Installing {w} from fresh built package.".format(w=built_package))
 
+            if not args.pre_download:
+                requirements = get_install_requires(os.path.join(os.path.abspath(args.target_setup), "setup.py"))
+                azure_requirements = [req.split(";")[0] for req in requirements if req.startswith("azure")]
+
+                if azure_requirements:
+                    logging.info(
+                        "Found {} azure requirement(s): {}".format(len(azure_requirements), azure_requirements)
+                    )
+
+                    download_command = [
+                        sys.executable,
+                        "-m",
+                        "pip",
+                        "download",
+                        "-d",
+                        tmp_dl_folder,
+                        "--no-deps",
+                    ]
+
+                    download_command.extend(azure_requirements)
```

Comment:
Are we always downloading latest of required packages? Ideally that should be fine since dependency is against what's in repo unless some package decides to pin dependency. right?

Reply:
Correct! We actually explicitly _disable_ this piece in `latestdependency` and `mindependency`. My offline text chat describes the issue with those however.